### PR TITLE
Address #36 and #37

### LINKED ISSUE #36: Describe schema for locale data
Most Intl types have a section under internal slots of the constructor that describes how the locale data is organized. E.g., see [this section for DateTimeFormat](https://tc39.es/ecma402/#sec-intl.datetimeformat-internal-slots). I don't see this description for DisplayNames.

### PR DESCRIPTION
**Pull request by FrankYFTang**
Improve text in the internal slot and use the right syntax.
This should close #36 and #37

### CODE REVIEW

File: index.html
Comment (ljharb) [inline index.html]:
```suggestion
        <code>"language"</code>, <code>"region"</code>, <code>"script"</code>, <code>"currency"</code>, <code>"dateField"</code>, and <code>"dateSymbol"</code>.
```

File: index.html
Comment (ljharb) [inline index.html]: is “value” typically capitalized like this in 402?

File: spec.emu
Comment (ljharb) [inline spec.emu]:
```suggestion
        `"language"`, `"region"`, `"script"`, `"currency"`, `"dateField"`, and `"dateSymbol"`.
```

**Comment by FrankYFTang** [comment]
@sffc

**Comment by sffc** [comment]
I'm seeing changes from #43 in this PR.  Can you check that you merged/rebased your branch correctly?

**Comment by FrankYFTang** [comment]
PTAL

File: spec.emu
Comment (sffc) [inline spec.emu]: I think you should use `&gt;` instead of raw `<`.  At least that's how the numberformat spec works.

File: spec.emu
Comment (sffc) [inline spec.emu]: I think `Record` is supposed to always be titlecase.

File: spec.emu
Comment (sffc) [inline spec.emu]: Please describe in more detail what the fields in this record represent.  I assume they are the identifiers that you pass to the `.of` method?  Do you have language for what you call those identifiers?

**Review by sffc — CHANGES_REQUESTED** [review]

File: spec.emu
Comment (FrankYFTang) [reply spec.emu]: done

File: spec.emu
Comment (FrankYFTang) [reply spec.emu]: done

File: spec.emu
Comment (FrankYFTang) [reply spec.emu]: done

**Comment by FrankYFTang** [comment]
PTAL

**Review by sffc — CHANGES_REQUESTED** [review]
Please send another PR converting the identifier lists into tables, like this one in NumberFormat:

https://github.com/tc39/proposal-unified-intl-numberformat/pull/58

Then, in this section, instead of listing out all the identifiers again, just put a reference to the table.

**Comment by FrankYFTang** [comment]
> Please send another PR converting the identifier lists into tables, like this one in NumberFormat:
> 
> [tc39/proposal-unified-intl-numberformat#58](https://github.com/tc39/proposal-unified-intl-numberformat/pull/58)
> 
> Then, in this section, instead of listing out all the identifiers again, just put a reference to the table.

It is already in https://github.com/tc39/proposal-intl-displaynames/pull/44 5 days ago but I cannot refer to that until that got merged.

**Comment by leobalter** [comment]
I just reviewed #44, this one also looks good. I wonder if Shane's request can be addressed in a follow up PR? I always prefer the granular patches. Makes it easier to review smaller changes.

**Comment by sffc** [comment]
Thanks for the pointer to #44!  As long as Frank follows up soon with a PR to refer to the tables, I'm okay with merging this PR, but I don't see a reason why not to wait until #44 is merged and then we just refer to the tables from here.  The part of the spec where Frank is duplicating all the codes is a big part of what this PR is adding.  If that text will be obsolete in 24 hours, let's wait 24 hours and then update this PR to refer to the new shiny tables.

**Comment by FrankYFTang** [comment]
ok, I merge the table change in and also move the two tables down and remove two tiny abstract operation which no longer make sense to exit. Also referring to that two tables from the description of the internal slot and add Description column to these two tables.

File: spec.emu
Comment (sffc) [inline spec.emu]: according to... what? (here and elsewhere)

File: spec.emu
Comment (sffc) [inline spec.emu]: Your prose is wrapped to inconsistent line width.  I suggest just not hard-wrapping your lines.  This is how most of the spec deals with long paragraphs (just all on one line).

File: spec.emu
Comment (sffc) [inline spec.emu]: Please use Markdown instead of `<ul>`/`<li>` in .emu files.

File: spec.emu
Comment (sffc) [inline spec.emu]: Is there a way you can avoid duplicating the three display name styles in each list item?  Something along the lines of,

> 1. ...
> 1. The value of this field must be a Record, which must have fields with the names of the six display name types: `"language"`, `"region"`, `"script"`, `"currency"`, `"dateTime"`, and `"dateTimeField"`.
> 1. The value of each of those fields in turn must be a Record that has fields corresponding to the three display name styles: `"narrow"`, `"short"`, and `"long"`.
> 1. The three display name styles fields under display name type `"language"` should contain Records, with keys corresponding to language codes according to ___. The value of these fields must be string values.
> 1. ...

**Review by sffc — CHANGES_REQUESTED** [review]

**Comment by FrankYFTang** [comment]
PTAL

**Review by sffc — COMMENTED** [review]
LGTM; would like one of the editors to also look before merging.

**Comment by FrankYFTang** [comment]
@zenparsing @ljharb

File: spec.emu
Comment (ljharb) [inline spec.emu]: this seems like it should be a named record type, not just a generic "Record"?

File: spec.emu
Comment (ljharb) [inline spec.emu:368]: same here, and throughout

File: spec.emu
Comment (ljharb) [inline spec.emu:425]: is the slot name `<code>`?

Comment (FrankYFTang) [reply spec.emu:425]: no, the variable is code, as defined in "Let _code_ be ? ToString(_code_) 3 steps above.

File: spec.emu
Comment (FrankYFTang) [reply spec.emu]: could you point me to an example in either ECMA262 or ECMA402 that I can follow?

File: spec.emu
Comment (FrankYFTang) [reply spec.emu:368]: could you point me to an example in either ECMA262 or ECMA402 that I can follow?

File: spec.emu
Comment (ljharb) [reply spec.emu:425]: ah, it's a dynamic lookup. Is that done anywhere else? Typically slot names are exclusively static/hardcoded.

File: spec.emu
Comment (ljharb) [reply spec.emu]: https://tc39.es/ecma262/#sec-chosen-value-records, https://tc39.es/ecma262/#script-record, etc - we do indeed have some generic records, but it seems much more readable to have a linkable section with a record type instead of effectively passing around a bunch of objects.

Comment (sffc) [reply spec.emu]: The pattern Frank followed here is pretty much consistent with how other ECMA-402 `[[LocaleData]]` objects are handled:

- [Intl.NumberFormat](https://www.ecma-international.org/ecma-402/6.0/index.html#sec-intl.numberformat-internal-slots)
- [Intl.DateTimeFormat](https://www.ecma-international.org/ecma-402/6.0/index.html#sec-properties-of-intl-datetimeformat-instances)

It's messy, but consistent.  If we clean it up, I'd rather do it across the spec.  I'm hoping to partially address this in tc39/ecma402#210

File: spec.emu
Comment (sffc) [reply spec.emu:425]: It's a common idiom in `[[LocaleData]]` slots in ECMA-402.

File: spec.emu
Comment (FrankYFTang) [reply spec.emu]: so I assume  this comment is addressed. Open another Issue if we still need to do that.

**Comment by FrankYFTang** [comment]
so should I consider all the issues addressed for this PR?

File: spec.emu
Comment (sffc) [inline spec.emu]: According to tc39/ecma402#360, maybe this should be `a [[types]] field` instead of `a types field`.  @Ms2ger

Comment (sffc) [reply spec.emu]: Also @leobalter @spectranaut

Comment (Ms2ger) [reply spec.emu]: I think so, yes; see also https://github.com/tc39/ecma402/pull/350.

**Comment by FrankYFTang** [comment]
PTAL

**Review by sffc — APPROVED** [review]
LGTM from me, with the expectation that there will be a follow-up PR to split dateTime into four separate types as discussed today in ECMA-402 meeting.

**Pull request merged**
